web-prover-core/src/manifest.rs

```diff
@@ -0,0 +1,256 @@
+use derive_more::From;
+use serde::{Deserialize, Serialize};
+use tiny_keccak::{Hasher, Keccak};
+
+use crate::{
+  errors::ManifestError,
+  http::{ManifestRequest, ManifestResponse},
+};
+
+/// Manifest containing [`ManifestRequest`] and [`ManifestResponse`]
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, From)]
+// #[serde(rename_all = "camelCase")]
+pub struct Manifest {
+  // /// Manifest version
+  // pub manifest_version: String,
+  // /// ID of the manifest
+  // pub id:               String,
+  // /// Title of the manifest
+  // pub title:            String,
+  // /// Description of the manifest
+  // pub description:      String,
+  /// HTTP request lock items
```

Comment:
nit: are these needed? can we remove it?

Reply:
It wasn't wise of me to leave a commented-out code in this PR, but I was hoping to address this in the "rich manifest PR" I was cooking